bpf/enhancedrecording/command.bpf.c

```diff
@@ -42,9 +44,20 @@ BPF_COUNTER(lost);
 
 static int __submit_arg(void *ptr, struct data_t *data)
 {
-    bpf_probe_read_user(data->argv, sizeof(data->argv), ptr);
-    if (bpf_ringbuf_output(&execve_events, data, sizeof(struct data_t), 0) != 0)
-        INCR_COUNTER(lost);
+    long read = 0;
+    long total_read = 0;
+    do {
+        read = bpf_probe_read_user_str(data->argv, sizeof(data->argv), (void *)ptr+total_read);
```

Comment:
According to the reference that I found (https://man7.org/linux/man-pages/man7/bpf-helpers.7.html), the `bpf_probe_read_user_str` function can return a negative value in case of error. Looks like we don't check for it. This was already a problem with the previous code, but it would be nice to fix it along with this modification. I see that the `__submit_arg` function has been designed to return an error code, but we just don't use this possibility (and we don't check for it in line 100: `__submit_arg((void *)filename, &data);`).

There's also a minor optimization that we might want to apply here: if the number of bytes fills precisely the number of bytes in the buffer, the next `bpf_probe_read_user_str` will return 0, in which case we'd send an unnecessary message with an empty `argv` string. (I'm not sure, though, whether it's really worth optimizing, unless we are really short on the number of items in the ring buffer.)